### PR TITLE
Adding some fixes for WOQL treatment of documents

### PR DESCRIPTION
**Pull request by GavinMendelGleason**

### CODE REVIEW

File: src/core/query/woql_compile.pl [hunk 1]
```diff
@@ -1546,7 +1542,7 @@
     compile_arith(Arith,Pre_Term,ArithE).
 compile_wf(dot(Dict,Key,Value), get_dict(KeyE,DictE,ValueE)) -->
     resolve(Dict,DictE),
-    resolve(Key,KeyE),
+    {atom_string(KeyE,Key)},
```
Comment (spl) [inline src/core/query/woql_compile.pl:1545]: What is the difference between `{atom_string(KeyE,Key)}` and `atom_string(KeyE,Key)`?

**Comment by spl** [comment]
Do we need to update the clients for this change?

File: src/core/query/woql_compile.pl (hunk 1 above)
Comment (GavinMendelGleason) [reply src/core/query/woql_compile.pl:1545]: We are in a DCG so there are two implicit parameters unless we wrap in a `{ }` block

**Comment by GavinMendelGleason** [comment]
Yes, but I don't think there is any regression as the clients are not correctly implementing the document interface in WOQL yet. We need to create integration tests in both.

**Pull request merged**